qiskit/quantum_info/primitives/estimator/pauli_estimator.py

```diff
@@ -0,0 +1,331 @@
+                            meas_circuit.sdg(i)
+                        if val in ["Y", "X"]:
+                            meas_circuit.h(i)
+                        meas_circuit.measure(i, i)
+                    coeff = coeff.real.item() if np.isreal(coeff) else coeff.item()
+                    meas_circuit.metadata = {"basis": pauli, "coeff": coeff}
+                    diff_circuits.append(meas_circuit)
+
+            preprocessed_circuits.append((circuit.copy(), diff_circuits))
+        return preprocessed_circuits
+
+    def _postprocessing(self, result: Union[Result, BaseResult, dict]) -> EstimatorResult:
+        """
+        Postprocessing for evaluation of expectation value using pauli rotation gates.
+        """
+        if not isinstance(result, SamplerResult):
+            raise TypeError(f"result must be SamplerResult, not {type(result)}.")
+
+        data = result.quasi_dists
+        metadata = result.metadata
+
+        combined_expval = 0.0
+        combined_variance = 0.0
+        combined_stderr = 0.0
+
+        for datum, meta in zip(data, metadata):
+            basis = meta.get("basis", None)
+            coeff = meta.get("coeff", 1)
+            basis_coeff = coeff if isinstance(coeff, dict) else {basis: coeff}
+            for basis, coeff in basis_coeff.items():
+                diagonal = _pauli_diagonal(basis) if basis is not None else None
+                # qubits = meta.get("qubits", None)
+                shots = sum(datum.values())
+
+                # Compute expval component
+                expval, var = _expval_with_variance(datum, diagonal=diagonal)
+                # Accumulate
+                combined_expval += expval * coeff
+                combined_variance += var * coeff**2
+                combined_stderr += np.sqrt(max(var * coeff**2 / shots, 0.0))
+
+        return EstimatorResult(
+            np.array([combined_expval], np.float64),
+            np.array([combined_variance], np.float64),
+        )
+
+
+def _expval_with_variance(
+    quasi: QuasiDistribution,
+    diagonal: Optional[np.ndarray] = None,
+    # clbits: Optional[list[int]] = None,
+) -> tuple[float, float]:
+
+    # Marginalize counts
+    # if clbits is not None:
+    #    counts = marginal_counts(counts, meas_qubits=clbits)
+
+    probs = np.fromiter(quasi.values(), dtype=float)
+    shots = probs.sum()
+    probs = probs / shots
+
+    # Get diagonal operator coefficients
+    if diagonal is None:
+        coeffs = np.array(
+            [(-1) ** (bin(key).count("1") % 2) for key in quasi.keys()], dtype=probs.dtype
+        )
+    else:
+        coeffs = np.asarray(diagonal[list(quasi.keys())], dtype=probs.dtype)
+
+    # Compute expval
+    expval = coeffs.dot(probs)
+
+    # Compute variance
+    if diagonal is None:
+        # The square of the parity diagonal is the all 1 vector
+        sq_expval = np.sum(probs)
+    else:
+        sq_expval = (coeffs**2).dot(probs)
+    variance = sq_expval - expval**2
```

Comment:
This is not a correct formula if someone is using M3.  To get it using M3 one must request the mitigation overhead: https://qiskit.org/documentation/partners/mthree/error.html,

Reply:
Thanks. I notice a PR to integrate M3 as a mitigator class is ongoing #7595. So, I'm thinking of replacing the mitigation part and the backend part of our code with the mitigator class.